samples/bluetooth/peripheral_uart/src/main.c

```diff
@@ -218,6 +240,36 @@ static int uart_init(void)
 		return -ENXIO;
 	}
 
+	if (IS_ENABLED(CONFIG_USB)) {
+		err = usb_enable(NULL);
+		if (err != 0) {
+			LOG_ERR("Failed to enable USB");
+			return err;
+		}
+
+		LOG_INF("Wait for DTR");
```

Comment:
If we start transmission before the CDC is ready, we would never get interrupt from it and the buffer would overflow in the end. This is something that probably should be reviewed and fixed in the adapter itself. Now the part you are referencing to is enabled when USB is enabled. But it might be the best to enable it if CDC_ACM was selected or the adapter is used?

Reply:
I was able to fix the buffer blocked issue - the tx is disabled to early. But if we do not wait for DTR, the welcome message would be lost.

Reply:
It makes the most sense to wait for DTR and set DTR if CONFIG_UART_LINE_CTRL is enabled.